benchmark/matrix_statistics/matrix_statistics.cpp

```diff
@@ -149,72 +132,90 @@ void extract_matrix_statistics(gko::matrix_data<etype, gko::int64>& data,
         ++col_dist[v.column];
     }
 
-    add_or_set_member(problem, "rows", data.size[0], allocator);
-    add_or_set_member(problem, "columns", data.size[1], allocator);
-    add_or_set_member(problem, "nonzeros", data.nonzeros.size(), allocator);
+    problem["rows"] = data.size[0];
+    problem["columns"] = data.size[1];
+    problem["nonzeros"] = data.nonzeros.size();
 
     std::sort(begin(row_dist), end(row_dist));
-    add_or_set_member(problem, "row_distribution",
-                      rapidjson::Value(rapidjson::kObjectType), allocator);
-    compute_distribution_properties(row_dist, problem["row_distribution"],
-                                    allocator);
+    problem["row_distribution"] = json::object();
+    compute_distribution_properties(row_dist, problem["row_distribution"]);
 
     std::sort(begin(col_dist), end(col_dist));
-    add_or_set_member(problem, "col_distribution",
-                      rapidjson::Value(rapidjson::kObjectType), allocator);
-    compute_distribution_properties(col_dist, problem["col_distribution"],
-                                    allocator);
+    problem["col_distribution"] = json::object();
+    compute_distribution_properties(col_dist, problem["col_distribution"]);
 }
 
 
-int main(int argc, char* argv[])
-{
-    std::string header =
-        "A utility that collects additional statistical properties of the "
-        "matrix.\n";
-    std::string format = example_config;
-    initialize_argument_parsing(&argc, &argv, header, format);
+using Generator = DefaultSystemGenerator<etype, gko::int64>;
 
-    std::clog << gko::version_info::get() << std::endl;
 
-    rapidjson::IStreamWrapper jcin(get_input_stream());
-    rapidjson::Document test_cases;
-    test_cases.ParseStream(jcin);
-    if (!test_cases.IsArray()) {
-        print_config_error_and_exit();
-    }
+struct MatrixStatistics : Benchmark<int> {
```

Comment:
The state of this benchmark is not used, right? So perhaps an empty struct would be nicer.